src/pages/LoginPage.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useRef } from 'react';
+import {
+  TextField,
+  Button,
+  Box,
+  Paper,
+  Typography,
+  IconButton,
+  InputAdornment,
+  InputLabel,
+  OutlinedInput,
+  FormControl
+} from '@mui/material';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+
+const LoginPage: React.FC = () => {
+  const [showPassword, setShowPassword] = useState<boolean>(false);
+  const [email, setEmail] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const passwordInputRef = useRef<HTMLInputElement>(null);
+
+  const handleClickShowPassword = () => {
+    if (passwordInputRef.current) {
+      const cursorPosition = passwordInputRef.current.selectionStart;
+      setShowPassword((prev) => !prev);
+      setTimeout(() => {
+        if (passwordInputRef.current) {
+          passwordInputRef.current.setSelectionRange(cursorPosition!, cursorPosition!);
+        }
+      }, 0);
+    }
+  };
```

Comment:
Why did you choose to do it this way when there is even a simpler implementation in the official documentation? [here](https://mui.com/material-ui/react-text-field/#system-InputAdornments.tsx)